sklearn/linear_model/base.py

```diff
@@ -143,13 +140,81 @@ def center_data(X, y, fit_intercept, normalize=False, copy=True,
     return X, y, X_mean, y_mean, X_std
 
 
+def _preprocess_data(X, y, fit_intercept, normalize=False, copy=True,
+                     sample_weight=None, return_mean=False):
+    """
+    Centers data to have mean zero along axis 0. If fit_intercept=False or if
+    the X is a sparse matrix, no centering is done, but normalization can still
+    be applied. The function returns the statistics necessary to reconstruct
+    the input data, which are X_mean, y_mean, X_norm, such that the output X'
+
+        X ' = (X - X_mean) / X_norm
```

Comment:
as X_mean can be 0 ie not be the mean I suggest we call it differently. How about X_offset and X_scale ?

then we can return X_offset and X_mean where X_offset can be zeroes and X_mean is the real mean. I think this X_mean which can be zero when fit_intercept is False is very confusing.

I plead guilty of course...